blocks/library/button/index.js

```diff
@@ -32,7 +32,8 @@ registerBlock( 'core/button', {
 	attributes: {
 		url: attr( 'a', 'href' ),
 		title: attr( 'a', 'title' ),
-		text: children( 'a' )
+		text: children( 'a' ),
+		align: metadata()
```

Comment:
I'm personally on the fence, but how do you feel about being even more explicit here with defining the metadata key to be extracted, rather than inferred by the key of the attribute.

For example:

```js
align: metadata( 'align' )
```

My worry with this would be that there'd likely be very undesirable consequences if the developer were to make a typo or otherwise use an argument different from the key.